File: tests/smoke/MainForm.pas
unit MainForm;

interface

uses
  System.SysUtils, System.Classes, Vcl.Controls, Vcl.Forms, Vcl.StdCtrls;

type
  TForm1 = class(TForm)
    Button1: TButton;
    Label1: TLabel;
    procedure Button1Click(Sender: TObject);
  private
    function Total(Count: Integer): Integer;
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

function TForm1.Total(Count: Integer): Integer;
var
  Index: Integer;
begin
  Result := 0;
  for Index := 1 to Count do
    Result := Result + Index;
end;

procedure TForm1.Button1Click(Sender: TObject);
var
  Answer: Integer;
begin
  Answer := Total(10);
  Label1.Caption := 'Total=' + IntToStr(Answer);
end;

end.

File: tests/smoke-cpp/MainForm.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "MainForm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
int __fastcall TForm1::Total(int Count)
{
	int Result = 0;
	for (int Index = 1; Index <= Count; Index++)
		Result += Index;
	return Result;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button1Click(TObject *Sender)
{
	int Answer = Total(10);
	Label1->Caption = "Total=" + IntToStr(Answer);
}
//---------------------------------------------------------------------------
